infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -118,11 +121,19 @@
      *
      * {@link SequencerServer::maxConflictWildcard} :
      * a "wildcard" representing the maximal update timestamp of
-     * all the confict keys which were evicted from the cache
+     * all the conflict keys which were evicted from the cache
+     *
+     * * {@link SequencerServer::maxConflictNewSequencer} :
+     * represents the max update timestamp of all the conflict keys
+     * which were evicted from the cache by the time this server is elected
+     * the primary sequencer. This means that any snapshot timestamp below this
+     * actual threshold would abort due to NEW_SEQUENCER cause.
      */
+    private final Cache<String, Long> conflictToGlobalTailCache;
+
     private long maxConflictWildcard = Address.NOT_FOUND;
 
-    private final Cache<String, Long> conflictToGlobalTailCache;
+    private long maxConflictNewSequencer = Address.NOT_FOUND;
```

Comment:
This name can be improved, maybe make initialToken (from the bootstrap method) global and re-use it to detect timestamps lower than that?